diff --git a/drivers/mmc/exynos_dw_mmc.c b/drivers/mmc/exynos_dw_mmc.c
index 7ccd113bd79..f1955313017 100644
--- a/drivers/mmc/exynos_dw_mmc.c
+++ b/drivers/mmc/exynos_dw_mmc.c
@@ -195,20 +195,7 @@ static unsigned int exynos_dwmci_get_clk(struct dwmci_host *host, uint freq)
 	u8 clk_div;
 	int err;
 
-	/* Should be double rate for DDR or HS mode */
-	if ((host->mmc->selected_mode == MMC_DDR_52 &&
-	     host->mmc->bus_width == 8) ||
-	    host->mmc->selected_mode == MMC_HS_400) {
-		freq *= 2;
-	}
-
 	clk_div = exynos_dwmmc_get_ciu_div(host);
-	err = exynos_dwmmc_set_sclk(host, freq * clk_div);
-	if (err) {
-		printf("DWMMC%d: failed to set clock rate (%d); "
-		       "continue anyway\n", host->dev_index, err);
-	}
-
 	err = exynos_dwmmc_get_sclk(host, &sclk);
 	if (err) {
 		printf("DWMMC%d: failed to get clock rate (%d)\n",
